src/metadata/debug_utils.rs

```diff
@@ -225,6 +227,45 @@ impl DebugUtils {
         Ok(())
     }
 
+    pub fn print_felt252<'c, 'a>(
+        &mut self,
+        context: &'c Context,
+        module: &Module,
+        block: &'a Block<'c>,
+        value: Value<'c, '_>,
+        location: Location<'c>,
+    ) -> Result<()>
+    where
+        'c: 'a,
+    {
+        if self.active_map.insert(DebugBinding::PrintFelt252) {
+            module.body().append_operation(func::func(
+                context,
+                StringAttribute::new(context, "__debug__print_felt252"),
+                TypeAttribute::new(
+                    FunctionType::new(context, &[IntegerType::new(context, 252).into()], &[])
+                        .into(),
+                ),
+                Region::new(),
+                &[(
+                    Identifier::new(context, "sym_visibility"),
+                    StringAttribute::new(context, "private").into(),
+                )],
+                Location::unknown(context),
+            ));
+        }
+
+        block.append_operation(func::call(
+            context,
+            FlatSymbolRefAttribute::new(context, "__debug__print_felt252"),
+            &[value],
+            &[],
+            location,
+        ));
```

Comment:
We should change it so that MLIR emits 4 arguments, or change the implementation to a pointer so that we can be 100% sure it'll work anywhere.